build_tools/cmake/iree_mlir_benchmark_suite.cmake

```diff
@@ -81,53 +82,82 @@ function(iree_mlir_benchmark_suite)
-    #
-    # Note: We actually should not do this; instead, we should directly compile
-    # from the initial source (i.e., TensorFlow Python models). But that is
-    # tangled with the pending Python testing infrastructure revamp so we'd
-    # prefer to not do that right now.
-    if("${_MODULE_MLIR_SOURCE}" MATCHES "^https?://")
+    if("${_MODULE_SOURCE}" MATCHES "^https?://")
+      set(_SOURCE_URL "${_MODULE_SOURCE}")
       # Update the source file to the downloaded-to place.
-      string(REPLACE "/" ";" _SOURCE_URL_SEGMENTS "${_MODULE_MLIR_SOURCE}")
+      string(REPLACE "/" ";" _SOURCE_URL_SEGMENTS "${_SOURCE_URL}")
       list(POP_BACK _SOURCE_URL_SEGMENTS _LAST_URL_SEGMENT)
-      set(_DOWNLOAD_TARGET_NAME "iree-download-benchmark-source-${_LAST_URL_SEGMENT}")
+      set(_DOWNLOAD_TARGET "iree-download-benchmark-source-${_LAST_URL_SEGMENT}")
 
-      string(REPLACE "tar.gz" "mlir" _FILE_NAME "${_LAST_URL_SEGMENT}")
-      set(_SOURCE_FILE "${_ROOT_ARTIFACTS_DIR}/${_MODULE_NAME}.mlir")
-
-      if (NOT TARGET "${_DOWNLOAD_TARGET_NAME}")
+      # Strip off gzip suffix if present (downloader unzips if necessary)
+      string(REGEX REPLACE "\.gz$" "" _SOURCE_FILE_BASENAME "${_LAST_URL_SEGMENT}")
+      set(_MODULE_SOURCE "${_ROOT_ARTIFACTS_DIR}/${_SOURCE_FILE_BASENAME}")
+      if (NOT TARGET "${_DOWNLOAD_TARGET}")
         add_custom_command(
-          OUTPUT "${_SOURCE_FILE}"
+          OUTPUT "${_MODULE_SOURCE}"
           COMMAND
             "${Python3_EXECUTABLE}" "${IREE_ROOT_DIR}/scripts/download_file.py"
-            "${_MODULE_MLIR_SOURCE}" -o "${_ROOT_ARTIFACTS_DIR}"
+            "${_SOURCE_URL}" -o "${_MODULE_SOURCE}"
           DEPENDS
             "${IREE_ROOT_DIR}/scripts/download_file.py"
-          COMMENT "Downloading ${_MODULE_MLIR_SOURCE}"
+          COMMENT "Downloading ${_SOURCE_URL}"
+        )
+        add_custom_target("${_DOWNLOAD_TARGET}"
+          DEPENDS "${_MODULE_SOURCE}"
+        )
+      endif()
+      set(_MODULE_SOURCE_TARGET "${_DOWNLOAD_TARGET}")
+    endif()
+
+    # If the source is a TFLite file, import it.
+    if("${_MODULE_SOURCE}" MATCHES "\.tflite$")
+      if (NOT IREE_BUILD_TFLITE_COMPILER)
+        message(WARNING "Skipping benchmark for ${_MODULE_SOURCE} because"
+                        " IREE_BUILD_TFLITE_COMPILER if OFF")
+        continue()
+      endif()
+      set(_TFLITE_FILE "${_MODULE_SOURCE}")
+      set(_MODULE_SOURCE "${_TFLITE_FILE}.mlir")
+      get_filename_component(_TFLITE_FILE_BASENAME "${_TFLITE_FILE}" NAME)
+      set(_TFLITE_IMPORT_TARGET "iree-import-tflite-${_TFLITE_FILE_BASENAME}")
+      if (NOT TARGET "${_TFLITE_IMPORT_TARGET}")
+        add_custom_command(
+          OUTPUT "${_MODULE_SOURCE}"
+          COMMAND
+            "$<TARGET_FILE:integrations::tensorflow::iree_tf_compiler::iree-import-tflite>"
+            "${_TFLITE_FILE}"
+            "-o=${_MODULE_SOURCE}"
+          DEPENDS
+            "${_TFLITE_FILE}"
```

Comment:
I think this might be one of those cases where CMake automatically adds the dependency, but I never remember what those are, so yeah let's just make it explicit.